stl/src/syserror_import_lib.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+#include <__msvc_system_error_abi.hpp>
+
+#include <Windows.h>
+
+namespace {
+
+    struct _Whitespace_bitmap_t {
+        bool _Is_whitespace[256];
+
+        constexpr _Whitespace_bitmap_t() : _Is_whitespace{} {
+            _Is_whitespace[' ']  = true;
+            _Is_whitespace['\n'] = true;
+            _Is_whitespace['\r'] = true;
+            _Is_whitespace['\t'] = true;
+            _Is_whitespace['\0'] = true;
+        }
+
+        constexpr bool _Test(const char _Ch) const noexcept {
+            return _Is_whitespace[static_cast<unsigned char>(_Ch)];
+        }
+    };
+
+    constexpr _Whitespace_bitmap_t _Whitespace_bitmap;
+
+} // unnamed namespace
+
+extern "C" {
+unsigned long __CLRCALL_PURE_OR_STDCALL __std_get_string_size_without_trailing_whitespace(
+    const char* const _Str, unsigned long _Size) noexcept {
+    if (_Size != 0) {
+        do {
+            --_Size;
+        } while (_Whitespace_bitmap._Test(_Str[_Size]));
+    }
+
+    return _Size;
+}
+
+_NODISCARD size_t __CLRCALL_PURE_OR_STDCALL __std_system_error_allocate_message(
+    unsigned long _Message_id, _Out_ char** _Ptr_str) noexcept {

```

Comment:
Any reason `_Ptr_str` (and `_Str` on 66) can't be `const`? Or do we avoid top-level `const` on separately-compiled functions because of name mangling weirdness? (If so, why is `_Str` const on 37?)

Reply:
It can get top level const because this is an extern "C" function so we don't have to worry about the top level const name mangling bug.